src/main/java/com/societegenerale/commons/plugin/utils/ExclusionImportOption.java

```diff
@@ -1,22 +1,57 @@
 package com.societegenerale.commons.plugin.utils;
 
+import java.net.URI;
+
+import javax.annotation.Nonnull;
+
 import com.tngtech.archunit.core.importer.ImportOption;
 import com.tngtech.archunit.core.importer.Location;
 
 public class ExclusionImportOption implements ImportOption {
 
-    String patternToExclude;
+    private static final String CLASS = ".class";
+
+    private URI patternToExcludeDirectoryBased;
+
 
-    public ExclusionImportOption(String patternToExclude) {
-        this.patternToExclude=patternToExclude;
+    public ExclusionImportOption(@Nonnull final String patternToExclude) {
+        // assuming a directory based exclude path
+        this.patternToExcludeDirectoryBased = toUri(patternToExclude);
     }
 
-    @Override
-    public boolean includes(Location location) {
+    private URI toUri(@Nonnull final String excludePath){
+        boolean excludePathEndsWithDotClass = excludePath.endsWith(CLASS);
+        String stringToConvert = excludePath;
+
+        if(excludePath.endsWith(CLASS)) {
+            // remove .class for easier replacements
+            stringToConvert = excludePath.substring(0, excludePath.length() - CLASS.length());
+        }
 
-        if(location.contains(patternToExclude)){
-            return false;
+        // assuming a package based exclude path
+        if(stringToConvert.contains(".")){
+            stringToConvert = stringToConvert.replaceAll("\\.", "/");
         }
-        return true;
+        // assuming a directory based exclude path
+        else {
+            // replace backslash by slash
+            stringToConvert = stringToConvert.replaceAll("\\\\", "/");
+        }
+
+        if(excludePathEndsWithDotClass){
+            // appends again .class that we removed initially
+            stringToConvert = stringToConvert.concat(CLASS);
+        }
+
+        stringToConvert = stringToConvert.replaceAll("://*", ":/");
```

Comment:
what do we convert here ? when I comment this line, the test is still passing. Can you please add a test that shows in which case this is necessary ?

Reply:
I don't know if replaceAll("://*", ":/"); is really necessary. I found it in com.tngtech.archunit.core.importer.NormalizedUri line 32. There is this comment "// this is how getClass().getResource(..) returns". I have never seen an URL containing "://" but to be sure that the comparsion of URIs and NormalizedUri works right I've decided to do the same as NormalizedUri.

Reply:
I've removed it - but we can always add it again later if required